src/Domain/EmailAddress.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Embeddable;
+use MatchBot\Application\Assertion;
+
+/** @Embeddable */
+class EmailAddress
+{
+    /** @Column(type = "string") */
+    public readonly string $email;
+
+    private function __construct(
+        string $value
+    ){
+        $this->email = $value;
+        Assertion::email($this->email);
+    }
+
+    /**
+     * @param string $emailAddress - must be a valid email address.
+     */
+    public static function of(string $emailAddress): self
```

Comment:
having a constructor called `of` is conventional for value objects - see e.g. https://github.com/brick/date-time/blob/master/src/Year.php#L37